appserver/web/war-util/src/main/java/org/glassfish/web/loader/WebappClassLoader.java

```diff
@@ -2147,11 +2147,10 @@ protected void nullInstance(Object instance) {
             return;
         }
         Field[] fields = instance.getClass().getDeclaredFields();
-        for (int i = 0; i < fields.length; i++) {
-            Field field = fields[i];
+        for (Field field : fields) {
             int mods = field.getModifiers();
             if (field.getType().isPrimitive()
-                    || (field.getName().indexOf("$") != -1)) {
+                    || (field.getName().indexOf('$') != -1)) {
```

Comment:
This could be achieved using _contains_

Reply:
indexOf(char) is faster than contains, I did some benchmarking to see what was the faster method.
```
Benchmark                                          Mode  Cnt     Score      Error  Uni
IndexOfAndContainsPerfTest.containsString          avgt   30    17.515 ±    0.249  ns/op
IndexOfAndContainsPerfTest.containsStringIndirect  avgt   30    19.722 ±    0.187  ns/op
IndexOfAndContainsPerfTest.indexOfChar             avgt   30    15.080 ±    0.199  ns/op
IndexOfAndContainsPerfTest.indexOfCharIndirect     avgt   30    16.278 ±    0.591  ns/op
IndexOfAndContainsPerfTest.indexOfString           avgt   30    17.334 ±    0.326  ns/op
IndexOfAndContainsPerfTest.indexOfStringIndirect   avgt   30    20.242 ±    0.340  ns/op
IndexOfAndContainsPerfTest.indexSplit              avgt   30   149.581 ±    7.285  ns/
```